vue.json

```diff
@@ -0,0 +1,67 @@
+{
+  "extends": ["plugin:vue/vue3-essential"],
+  "parserOptions": {
+    "extraFileExtensions": [".vue"]
+  },
+  "plugins": ["vue"],
+  "rules": {
+    "vue/attributes-order": "error",
+    "vue/component-tags-order": "error",
+    "vue/no-lone-template": "error",
+    "vue/no-multiple-slot-args": "error",
+    "vue/no-v-html": "error",
+    "vue/order-in-components": "error",
+    "vue/this-in-template": "error",
+    "vue/attribute-hyphenation": "error",
+    "vue/component-definition-name-casing": "error",
+    "vue/first-attribute-linebreak": "error",
+    "vue/html-closing-bracket-spacing": "error",
+    "vue/html-end-tags": "error",
+    "vue/multiline-html-element-content-newline": "error",
+    "vue/mustache-interpolation-spacing": "error",
+    "vue/no-multi-spaces": "error",
+    "vue/no-spaces-around-equal-signs-in-attribute": "error",
+    "vue/no-template-shadow": "error",
+    "vue/one-component-per-file": "error",
+    "vue/prop-name-casing": "error",
+    "vue/require-default-prop": "error",
+    "vue/require-explicit-emits": "error",
+    "vue/require-prop-types": "error",
+    "vue/v-bind-style": "error",
+    "vue/v-on-event-hyphenation": "error",
+    "vue/v-on-style": "error",
+    "vue/v-slot-style": "error",
+    "vue/html-quotes": "error",
+
+    "vue/html-self-closing": [
+      "error",
+      {
+        "html": {
+          "void": "always"
+        }
+      }
+    ],
+
+    "max-len": "off"
```

Comment:
I think this should be moved to the `overrides` section.

And I would still add some limit, like 160, because 160+ is too long even when using Tailwind (it's better to refactor Tailwind classes in that case).

Reply:
Yes, I will move it to the ```overrides``` section.

I am quite new to Tailwind. Wouldn't refactoring multiple classes into a single one be against Tailwinds principle? Could you perhaps shed some light on this?

Reply:
I haven't used Tailwind actually, but I know you can [extract classes using `@apply`](https://tailwindcss.com/docs/reusing-styles#extracting-classes-with-apply).

But maybe having lines longer than 160 characters with tens of classes is OK with Tailwind...

Reply:
Hmm, I do dislike it as well. However, as stated [here](https://tailwindcss.com/docs/reusing-styles#avoiding-premature-abstraction), it does look like it's the way to go unfortunately. We could just keep ```max-lines``` at 120 and leave disabling it if using Tailwind to the user?

Reply:
Sure, let's keep 120.